File: 86.js
function fractionalKnapsack(values, weights, W){
    let items = values.map((v,i)=>[v/weights[i], v, weights[i]]);
    items.sort((a,b)=>b[0]-a[0]);
    let total = 0;
    for(let [ratio,v,w] of items){
        if(W == 0) break;
        let take = Math.min(w, W);
        total += take * v / w;
        W -= take;
    }
    return total;
}

// Driver code
let values = [60,100,120];
let weights = [10,20,30];
let W = 50;
console.log("Maximum value:", fractionalKnapsack(values, weights, W));

File: 86.py
def fractional_knapsack(values, weights, W):
    items = sorted(zip(values, weights), key=lambda x: x[0]/x[1], reverse=True)
    total = 0
    for v, w in items:
        if W == 0:
            break
        take = min(w, W)
        total += take * v / w
        W -= take
    return total

# Driver code
if __name__ == "__main__":
    values = [60, 100, 120]
    weights = [10, 20, 30]
    W = 50
    print("Maximum value:", fractional_knapsack(values, weights, W))
